### ISSUE TITLE
Webhook signature verification failed

### ISSUE THREAD

**Issue by deiks** [sendgrid/sendgrid-php#1032]
### Issue Summary
Hi there, trying to verify signature with PHP as in this example https://github.com/sendgrid/sendgrid-php/blob/main/examples/helpers/eventwebhook/example.php

Verification key is correct, signature, timestamp and payload matches to the original query.

I've seen that there's problem with missing trailing \r\n and tried with that as well but unsuccessfully.

``` 
$publicKeyECDSA = $webhook->convertPublicKeyToECDSA("MFkwEwYH....mfZx+dPg==");

$signature = $webhook->verifySignature(
                $publicKeyECDSA, 
                $request->getContent(),
                $request->header('X-Twilio-Email-Event-Webhook-Signature'), 
                $request->header('X-Twilio-Email-Event-Webhook-Timestamp')
);
```

**### Note: Using Laravel.**

### Technical details:
* sendgrid-php version: 7.9.2
* php version: 7.3

Can it be that verification key has to match server name (domain in webhook url) and not possible test locally?

**Comment by eshanholtz** [comment]
@deiks I'm not positive I understand what your last question is asking. Can you rephrase?

**Comment by deiks** [comment]
Ah.. ignore it. Stupid question anyway :)

**Comment by JenniferMah** [comment]
Hi @deiks I'm a bit unclear about what you need assistance with. Were you able to check for a valid signature following the [example](https://github.com/sendgrid/sendgrid-php/blob/main/examples/helpers/eventwebhook/example.php)?

**Comment by deiks** [comment]
Negative. It always reports false. Even if I pick correct data from call.

**Comment by shwetha-manvinkurke** [comment]
@deiks Can you show us the code sample for your `getContent()` and the content itself that you are feeding into `verifySignature`?

**Comment by deiks** [comment]
> @deiks Can you show us the code sample for your `getContent()` and the content itself that you are feeding into `verifySignature`?

I don't have it now, but it's payload that Sendgrid sent to my server, array of objects. And payload is matching signature and timestamp from headers. It's not like some dummy data.

**Comment by thinkingserious** [comment]
Hello @deiks,

When you obtain the payload data again, please send it along to us so that we may try to reproduce the error. Thank you!

With best regards,

Elmer

**Comment by deiks** [comment]
Ok, found one in logs. This is an example of payload;

```[
   {
      "email":"EMAiL_REMOVED",
      "event":"processed",
      "send_at":0,
      "sg_event_id":"cHJvY2Vzc...yNDdZR1RTZy0w",
      "sg_message_id":"zK_J6J2ESSuHrc247...8-60180BBE-BD.0",
      "smtp-id":"<zK_J6J2...@ismtpd0004p1lon1.sendgrid.net>",
      "timestamp":1612188606
   },
   {
      "email":"EMAIL_REMOVED",
      "event":"delivered",
      "ip":"x.x.x.x",
      "response":"250 2.0.0 36d0r3u5yb-1 Message accepted for delivery",
      "sg_event_id":"ZGVsaXZ...RVNTdUhyYzI0N1lHVFNnLTA",
      "sg_message_id":"zK_J6J2ES...d-ckfd8-18-60180BBE-BD.0",
      "smtp-id":"<zK_J6J2E...g@ismtpd0004p1lon1.sendgrid.net>",
      "timestamp":1612188609,
      "tls":1
   }
]
```

And I have 2 headers: **x-twilio-email-event-webhook-timestamp** and **x-twilio-email-event-webhook-signature**

So as in example in the first post, I tried to match as documented. But unsuccessfully. It's always false. And I can see that proper data (from webhook call) is available (payload, headers), so there is no something like some null values or something missing.

**Comment by thinkingserious** [comment]
Thanks @deiks!

In order to reproduce we will need the key and signature.

**Comment by doubbz** [comment]
hello,
I am interested in this issue, please don't close it without going through with it. We opened a similar issue (#1012) but couldn't go through with it because of priority concerns.. 
thank you in advance

**Comment by eshanholtz** [comment]
@doubbz If you could provide a test payload, timestamp, signature, and key so we can try and reproduce the error ourselves and track down that issue, that we greatly expedite us solving this!

**Comment by JenniferMah** [comment]
Closing this due to inactivity. Feel free to reopen if you still need help.

**Comment by ArtisanTinkerer** [comment]
I am getting the same problem:

```
$eventWebhook = new EventWebhook();
$ecPublicKey = $eventWebhook->convertPublicKeyToECDSA($publicKey));
$payload = json_encode( $request->all()[0] ) . "\r\n";

$valid = $eventWebhook->verifySignature(
            $ecPublicKey,
            $payload,
            $request->header('X-Twilio-Email-Event-Webhook-Signature'),
            $request->header('X-Twilio-Email-Event-Webhook-Timestamp')
        );

//payload:

{"email":"mr@yahoo.com","event":"processed","send_at":0,"sentMessage_id":890,"sg_event_id":"cHJvY2Vzc2VkLTE4MDY0NTQ4LU13d1RwSmpkVDlDTVdvSEZYbGtaM1EtMA","sg_message_id":"MwwTpJjdT9CMWoHFXlkZ3Q.filterdrecv-7bf5c69d5-jxm6x-1-6139EE27-31.0","smtp-id":"<add1a3b7b54ebc75341d8e6d8719bb8c@swift.generated>","tenant_id":69,"timestamp":1631186471}

//timestamp
1631186481

//signature
MEUCIFz7DRkkP8jzRwNyukVcjhNgvejZTRbX78v3pzkgyq9NAiEAuUOA07Xa4XNZmPuJ/qCVOCXjcl+NQoYaKzOlGRNxNXE=

```
